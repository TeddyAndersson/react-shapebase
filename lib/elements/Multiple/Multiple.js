import React from 'react';
import _ from 'lodash';
import styled, { css } from 'styled-components'
import PropTypes from 'prop-types';
import RectangleShape from '../Shape/RectangleShape';

class Element extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      activeOption: this.props.initialOption,
    }
  }

  handleClick = (id) => {
    this.setState({activeOption: id})

    _.invoke(this.props, 'onClick', id, this.props)
  }

  render(){
    const {
        className,
        options,
        activeBackground,
    } = this.props

    return (
        <ul className={className}>
          { _.values(options).map(option => (
            // Replace <li/> with a component for better state handling and generic code
                <li
                    key={option.id}
                    onClick={() => this.handleClick( option.id )}
                >
                    <RectangleShape
                        background={`${ option.id === this.state.activeOption ? activeBackground : 'rgba(0,0,0,0.2)'}`}
                        raised={option.id === this.state.activeOption ? 1 : 0}
                    >
                        <input
                        type="hidden"
                        name={option.name}
                        value={option.value}
                        autoComplete="off"
                        />
                        <div>
                            <span>
                                {option.label}
                            </span>
                        </div>
                    </RectangleShape>
                </li>
            ))}
        </ul>
    )
  };
};

Element.propTypes = {

    /**
      * The children of the component.
      */
    children: PropTypes.node,

    /**
      * Additonal classes.
      */
    className: PropTypes.string,

    /**
     * Description of the intreval the range.
     */
    description: PropTypes.bool,

    /**
     * Description of the intreval the range.
     */
    initialOption: PropTypes.string,

    /**
     * The range of the grades options 1-5 or 1-10
     */
    options: PropTypes.arrayOf(PropTypes.shape(
        {
            value: PropTypes.string,
            label: PropTypes.string,
            name: PropTypes.string,
            id: PropTypes.string,

        }
    )),
    /**
     * The background for an active option
     */
    activeBackground: PropTypes.string,
}

Element.defaultProps = {
    initialOption: undefined,
    activeBackground: "#1FAACB",
}

const Multiple = styled(Element)`
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 0;
    list-style: none;
    display: flex;
    margin: 0;
    font-weight: bold;
    font-size: 0.9em;
    color: ${props => props.color};
    
    li {
        margin-bottom: 0.4em
        div {
            width: 100%
            max-width: 100%;
            display: flex;
            justify-content: center;
            align-items: center;
            transition: all 0.3s ease
            flex: 1;
        }
    }

    li:last-child {
        margin: 0;
    }
    
    ${props => props.range === 10 && css`
        font-size: 0.8em;
    `}
`;

export default Multiple;