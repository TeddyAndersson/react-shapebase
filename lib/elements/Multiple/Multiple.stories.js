import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions/dist/preview';
import Grid from '../Grid';
import Multiple from './Multiple';

const lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."

storiesOf('Multiple', module)
  .add('Multiple', () => (
    <Grid columns={1} rowGap={0.5}>
      <Multiple
        onClick={action('Multiple')}
        initalOption={'2'}
        options={[
          {
            id: '1',
            name: '17 eller yngre',
            value: '1',
            label: '17 eller yngre',
          },
          {
            id: '2',
            name: '18-24',
            value: '2',
            label: '18-24',
          },
          {
            id: '3',
            name: '25-31',
            value: '3',
            label: '25-31',
          },
          {
            id: '4',
            name: '32-38',
            value: '4',
            label: '32-38',
          },
        ]}
      />
    </Grid>
  ));