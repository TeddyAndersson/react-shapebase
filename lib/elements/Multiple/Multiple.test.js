import React from 'react';
import Multiple from './Multiple'
import renderer from 'react-test-renderer';

test('should render Multiple', () => {
  const component = renderer.create(
    <Multiple
      onClick={()=>{}}
      initalOption={'2'}
      options={[
        {
          id: '1',
          name: '17 eller yngre',
          value: '1',
          label: '17 eller yngre',
        },
        {
          id: '2',
          name: '18-24',
          value: '2',
          label: '18-24',
        },
        {
          id: '3',
          name: '25-31',
          value: '3',
          label: '25-31',
        },
        {
          id: '4',
          name: '32-38',
          value: '4',
          label: '32-38',
        },
      ]}
    />  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
