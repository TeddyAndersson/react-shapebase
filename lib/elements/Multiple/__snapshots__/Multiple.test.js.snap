// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Multiple 1`] = `
<ul
  className="sc-htpNat eApmUK"
>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH lipQua sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name="17 eller yngre"
        type="hidden"
        value="1"
      />
      <div>
        <span>
          17 eller yngre
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH lipQua sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name="18-24"
        type="hidden"
        value="2"
      />
      <div>
        <span>
          18-24
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH lipQua sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name="25-31"
        type="hidden"
        value="3"
      />
      <div>
        <span>
          25-31
        </span>
      </div>
    </div>
  </li>
  <li
    onClick={[Function]}
  >
    <div
      className="sc-bwzfXH lipQua sc-bdVaJa jmjnGv"
    >
      <input
        autoComplete="off"
        name="32-38"
        type="hidden"
        value="4"
      />
      <div>
        <span>
          32-38
        </span>
      </div>
    </div>
  </li>
</ul>
`;
